include/triton/Dialect/TritonGPU/IR/TritonGPUAttrDefs.td

```diff
@@ -377,11 +382,16 @@ An encoding for tensors whose elements may be simultaneously accessed by
 different GPU threads in the programs, via shared memory. In other words,
 for all indices i \in Z^d, \mathcal{L}(i) = {0, 1, ..., 32*num_warps - 1}.
 Compared to SwizzledSharedEncodingAttr, this encoding uses padding to avoid
-shared memory bank conflicts.
+shared memory bank conflicts. For cases where padding alone cannot efficiently
+avoid shared memory bank conflicts it has an optional linear component to
+rearange elements. If the the linear component is absent the elements will be
+contiguous (identity mapping).
 
 Formally, given a layout:
     padded_shared<[<interval_0>:+<pad_0>, <interval_1>:+<pad_1>, ...]>
 We insert a padding of `<pad_i>` elements after every `<interval_i>` elements.
+The padding is computed based on the final element index after the optional
+linear component is applied to the element inidices.
```

Comment:
I think this wording is confusing. We have relatively two "disjoint" mechanisms here--1) the padding only concerns the final 1-D linearized memory allocation and padds certain amounts after certain elements. It has no concern over 2) the linear remapping, which concerns the 1-D shared memory offset to logical n-D tensor elements. The linear remapping applies assuming no padding and afterwards the padding applies ignoring how elements are rearranged? Can you refine the wording here to be clear?